doc/dev/release-checklists.rst

```diff
@@ -53,6 +53,13 @@ Misc
 - [x] update src/tools/monmaptool.cc (`min_mon_release` and corresponding output in `src/test/cli/monmaptool`)
 - [x] update src/cephadm/cephadm (`DEFAULT_IMAGE_RELEASE` to X)
 
+Docs
+~~~~
+
+- [x] Remove ``doc/releases/*.rst``.
```

Comment:
My immediate thought was "why not `doc/releases/*`?" and upon checking I noticed that `releases.yml` was retained in both quincy and pacific.  Why is that?

Reply:
It's used by some of the doc build machinery to tell if you're on a release branch, I think. I didn't study it very much and it doesn't hurt to leave it behind AFAICT.

Reply:
Can you add this explanation to the checklist item?  I bet the person doing this for reef or S is going to be confused with "why am I supposed to remove these rst files but not this yaml file that is bound to become stale just as rst files would".

Reply:
Yes, It's added now.

Reply:
Thanks!  I'm not sure I buy the "to tell if you're on a release branch" reasoning since all branches except for those that haven't been pushed to in years would have it, but that's beside the point.